core/src/components/rating/rating.ts

```diff
@@ -298,7 +320,47 @@ export function createRating(config?: PropsConfig<RatingProps>): RatingWidget {
 				}
 			},
 		},
-		directives: {},
+		directives: {
+			containerDirective: createAttributesDirective(() => ({
+				events: {
+					keydown: (e) => widget.actions.handleKey(e),
+					mouseleave: () => widget.actions.leave(),
+				},
+				attributes: {
+					role: 'slider',
+					class: className$,
+					'aria-valuemin': 0,
+					tabindex: tabindex$,
+					'aria-valuemax': maxRating$,
+					'aria-valuenow': visibleRating$,
+					'aria-valuetext': ariaValueText$,
+					'aria-readonly': computed(() => (readonly$() ? 'true' : undefined)),
+					'aria-disabled': computed(() => (disabled$() ? 'true' : undefined)),
+					'aria-label': ariaLabel$,
+					'aria-labelledby': computed(() => ariaLabelledBy$() || undefined),
+				},
+				classNames: {
+					'au-rating': writable(true),
+				},
+			})),
+			starDirective: createAttributesDirective((starContext$: ReadableSignal<{index: number}>) => {
+				const index = starContext$().index;
```

Comment:
`starContext$` is a readable signal, its value could change over time. If you evaluate it here, and the value changes later, the new value will not be used. You should evaluate `starContext$()` only when using it.